### PR TITLE
Adding Windows support and MacOS support

### PR DESCRIPTION
**Pull request by katmsft**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -106,36 +108,157 @@ set(AZURE_STORAGE_LITE_SOURCE
...
-
-if (NOT ${USE_OPENSSL})
-  find_package(GnuTLS REQUIRED)
-  set(EXTRA_INCLUDE ${GNUTLS_INCLUDE_DIR})
-  set(EXTRA_LIBRARIES ${GNUTLS_LIBRARIES})
-else()
+if(APPLE)
+  option(USE_OPENSSL "Use OpenSSL instead of GnuTLS" ON)
   add_definitions(-DUSE_OPENSSL)
+  find_package(CURL REQUIRED)
```
Comment (smtakeda) [inline CMakeLists.txt]: Can we add a custom version of Curl as well? Could be a separate commit/PR, though.
We use a custom version curl since we inject extra security checks.

File: CMakeLists.txt [hunk 2]
```diff
@@ -106,36 +108,157 @@ set(AZURE_STORAGE_LITE_SOURCE
...
+  endif()
+else()
+  # Add the parameter/definition specifically for Windows.
+  option(USE_OPENSSL "Use OpenSSL instead of GnuTLS" ON)
+  option(CMAKE_BUILD_TYPE "The default build type" "Debug")
+  option(CMAKE_GENERATOR_PLATFORM "The Generator platform used for CMake" "x86")
+  #Construct the libcurl folder path
+  #This is because the cmake's FindCurl.cmake not being compatible with libcurl's buildscript on Windows.
+  #e.g. libcurl-vc15-x64-debug-static-ssl-static-ipv6-sspi
+  if((DEFINED CURL_ROOT_DIR) AND (NOT (CURL_ROOT_DIR STREQUAL "")))
```
Comment (smtakeda) [inline CMakeLists.txt]: ok, this is valid only for Windows? could this be available for Mac and Linux too? There are two reasons why we want to have this:
- We have a custom curl library including: https://github.com/snowflakedb/libsnowflakeclient/blob/master/deps/curl-7.58.0/lib/vtls/sf_ocsp.c
- MacOS curl had a bug that could not enforce TLS 1.2

File: CMakeLists.txt [hunk 3]
```diff
@@ -106,26 +108,138 @@ set(AZURE_STORAGE_LITE_SOURCE
...
 
-if (NOT ${USE_OPENSSL})
-  find_package(GnuTLS REQUIRED)
-  set(EXTRA_INCLUDE ${GNUTLS_INCLUDE_DIR})
-  set(EXTRA_LIBRARIES ${GNUTLS_LIBRARIES})
+if (UNIX)
+  option(USE_OPENSSL "Use OpenSSL instead of GnuTLS" OFF)
+  find_package(Threads REQUIRED)
+  find_package(CURL REQUIRED)
+  pkg_search_module(UUID REQUIRED uuid)
```
Comment (rvishureddy) [inline CMakeLists.txt]: Does windows need external UUID package?
